azurelinuxagent/ga/exthandlers.py

```diff
@@ -871,6 +935,37 @@ def collect_ext_status(self, ext):
 
         return ext_status
     
+    def get_ext_handling_status(self, ext):
+        seq_no, ext_status_file = self.get_status_file_path(ext)
+        if seq_no < 0 or ext_status_file is None:
+            return None
+
+        # Missing status file is considered a non-terminal state here
+        # so that extension sequencing can wait until it becomes existing
+        if not os.path.exists(ext_status_file):
+            ext_status = ExtensionStatus(seq_no=seq_no)
```

Comment:
I think there no need to return an ExtensionStatus object. I think only the 'status' field is used by the code

Reply:
> I think only the 'status' field is used by the code

It will be easier to debug if we have the entire status object and not just the fields used by the code. Also, it is impossible to tell whether it is a status or a substatus otherwise.

Reply:
No more substatus per @narrieta 's suggestion.